### PR TITLE
Implemented functionality for diff --dest

### LINKED ISSUE #2103: swagger diff --dest="${DEST}": Is not writing output to "${DEST}"
## Problem statement

According to the documentation, https://github.com/go-swagger/go-swagger/blob/524d4f81cd9325321c441bac3b377831522d2a50/docs/diff/diff.md, `--dest=<destination_file>` should write output to `<destination_file>` unless I am misunderstanding it?

Looking at the code, https://github.com/go-swagger/go-swagger/blob/524d4f81cd9325321c441bac3b377831522d2a50/cmd/swagger/commands/diff.go#L40, I cannot see any other place `Destination` is being used.

Feel free to correct me, and many thanks.

## Swagger specification

https://github.com/banaio/openbanking_tools/tree/master/specifications/read-write

## Steps to reproduce

Could someone please confirm if a command like below should write the output to `/Users/mbana/work/openbankinguk/bitbucket/openbanking_tools/generated/diff/v3.1_account-info_vs_v4.0_account-info.json`:

```sh
$ swagger diff --log-output /Users/mbana/work/openbankinguk/bitbucket/openbanking_tools/generated/diff/v3.1_account-info_vs_v4.0_account-info.log -f json -d /Users/mbana/work/openbankinguk/bitbucket/openbanking_tools/generated/diff/v3.1_account-info_vs_v4.0_account-info.json
```

## Environment

```sh
$ swagger version 
dev
$ git --no-pager log -n2 
commit 55c2243453ca9e9e8f2edebbd9f50f06f50bb2ed (HEAD -> master, origin/master, origin/HEAD)
Author: Jim Turnquist <james.turnquist@emc.com>
Date:   Fri May 24 13:48:19 2019 -0500

    urlbuilder template fixed to support query params named 'append' (#1949)
    
    fixes #1948
    
    Update urlbuilder_test.go to support added query name mangling.
    
    Update bindata.go
    
    Signed-off-by: Jim Turnquist <jim@boonlogic.com>

commit 8ce1ff423c5a009e3707a030b69ea3f1b622a1ce
Merge: ffefbd27 973077dd
Author: Ivan Porto Carrero <ivan@flanders.co.nz>
Date:   Sun May 12 23:20:51 2019 -0700

    Merge pull request #1939 from tectonic-network/125-parameters-order
    
    support keep spec properties order
```

### PR DESCRIPTION
**Pull request by wjase**
* fixes #2103

### CODE REVIEW

**Comment by fredbi** [comment]
@wjase  here is the draft work I mentioned in my above comment:
https://github.com/go-swagger/go-swagger/pull/2332

Maybe you can get some inspiration from it.
Whatever, I appreciate your contribution and won't block your fix just because I picked the funny idea of refactoring tests :)

**Comment by wjase** [comment]
Righto thanks for the feedback. I'll take a look

File: cmd/swagger/commands/diff.go [hunk 1]
```diff
@@ -39,33 +39,51 @@ func (c *DiffCommand) Execute(args []string) error {
...
 	diffs, err := getDiffs(args[0], args[1])
 	if err != nil {
 		return err
 	}
 
-	ignores, err := readIgnores(c.IgnoreFile)
-	if err != nil {
-		return err
+	if len(c.IgnoreFile) > 0 {
+		log.Println("Read ignores")
```
Comment (fredbi) [inline cmd/swagger/commands/diff.go:49]: is it debug code or is there a reason for this extra verbosity? We already have a decent recap of what is going to happen when the process starts.

**Comment by fredbi** [comment]
@wjase  I believe this is now superseded by #1232. I've seen you had some additions with logging, though. What do you think is best from now on?

**Comment by wjase** [comment]
Hiya,
I'm happy if you just want to go with yours... Once that's in I'm chasing a bug with diff git headers.
Cheers
Jason

Sent from Yahoo7 Mail on Android 
 
  On Sun, 5 Jul 2020 at 6:34 pm, fredbi<notifications@github.com> wrote:   


@wjase I believe this is now superseded by #1232. I've seen you had some additions with logging, though. What do you think is best from now on?

**Comment by fredbi** [comment]
@wjase ok you may thus concentrate on chasing this bug, I took care of the "report file output" one.

A PR on that is always welcome. By the way, maybe it would be interesting to share the description of that bug.

You'll find it easier now to code tests on the output since you don't have to worry about capture.

Ultimately, I'd like the entire diff functionality to move to github.com/go-openapi/analysis, where it belongs,
rather than living here with the CLI, but this is not going to any time soon.

**Pull request closed without merging**